apis/v1alpha1/httproute_types.go

```diff
@@ -82,11 +82,28 @@ type HTTPRouteHost struct {
 	ExtensionRef *RouteHostExtensionObjectReference `json:"extensionRef" protobuf:"bytes,3,opt,name=extensionRef"`
 }
 
-// HTTPRouteRule is the configuration for a given path.
+// HTTPRouteRule defines semantics for matching an incoming HTTP requests against
+// a set of matching rules and executing an action (and optionally filters) on
+// the request.
 type HTTPRouteRule struct {
-	// Match defines which requests match this path.
+	// Matches define conditions used for matching the rule against
+	// incoming HTTP requests.
+	// Each match is independent, i.e. this rule will be matched
+	// if **any** one of the match is satisfied.
+	// For example, take the following matches configuration:
+	// matches:
+	// - path: /foo
+	//   headers:
+	//     version: "2"
+	// - path: /v2/foo
+	// For a request to match against this rule, a request should satisfy
+	// EITHER of the two conditions:
+	// - path prefixed with `/foo` AND contains the header `version: "2"`
+	// - path prefix of `/v2/foo`
+	// Please see doc for HTTPRouteMatch on how to specify multiple
+	// match conditions that should be ANDed together.
```

Comment:
This is the approach that makes the most sense to me - matching multiple paths or different values of the same header seems like it could be useful. I think others had mentioned that it felt inconsistent with other approaches and/or would be complex to implement. I can't quite remember who that feedback was from, maybe @jpeach or @bowei? Want to make sure this also makes sense to them.

Reply:
+1 Makes sense to me.